### DISCUSSION TITLE
Replace device

### DISCUSSION THREAD

**Discussion by BretOrrell** [netbox-community/netbox#15385] [Q&A]
We just replaced two of our routers with a new model, Is there any automated process to do a replace or migration from the old device type to the new one?

**Comment by candlerb** [comment]
No, but you can simply change the device type of an existing device to a new one, and then update its serial number, MAC addresses etc.

It won't affect any existing interfaces; you'll have to manually rename interfaces as required, and/or add new interfaces that the new device type has that the old one doesn't (there's a [script](https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py) that can help you with that)

The GUI has a bulk interface rename function if you want to rename all of GigabitEthernet0/1/x to GigabitEthernet1/1/x (or similar).

**Reply by BretOrrell** [reply]
Thanks.. I was afraid that was the answer.  We are still on an old version at the moment but was hopping that feature would have been added by now.
